src/JsonApiSerializer/JsonConverters/ResourceObjectConverter.cs

```diff
@@ -1,366 +1,377 @@
-        /// <returns></returns>
-        protected virtual string GenerateDefaultTypeName(Type type)
-        {
-            return type.Name.ToLowerInvariant();
-        }
-
-        internal string GenerateDefaultTypeNameInternal(Type type)
-        {
-            return GenerateDefaultTypeName(type);
-        }
-
-        /// <summary>
-        /// Exposes contract to allow overriding object initialisation based on resource type during deserialization.
-        /// </summary>
-        /// <param name="objectType">Type of the property that the created object will be assigned to</param>
-        /// <param name="jsonapiType">Type field specified on on the json api document</param>
-        /// <param name="serializer">The serializer.</param>
-        /// <returns></returns>
-        /// <exception cref="Exception"></exception>
-        protected virtual object CreateObject(Type objectType, string jsonapiType, JsonSerializer serializer)
-        {
-            var contract = serializer.ContractResolver.ResolveContract(objectType);
-            if (contract.DefaultCreator == null)
-            {
-                var typeInfo = objectType.GetTypeInfo();
-                if (typeInfo.IsInterface)
-                    throw new JsonSerializationException($"Could not create an instance of type {objectType}. Type is an interface and cannot be instantiated.");
-                else if (typeInfo.IsAbstract)
-                    throw new JsonSerializationException($"Could not create an instance of type {objectType}. Type is an abstract class and cannot be instantiated.");
-                else
-                    throw new JsonSerializationException($"Could not create an instance of type {objectType}.");
-            }
-
-            return contract.DefaultCreator();
-        }
-
-        internal object CreateObjectInternal(Type objectType, string jsonapiType, JsonSerializer serializer)
-        {
-            return this.CreateObject(objectType, jsonapiType, serializer);
-        }
-
-        
-    }
-}
+using JsonApiSerializer.ContractResolvers;
+using JsonApiSerializer.ContractResolvers.Contracts;
+using JsonApiSerializer.Exceptions;
+using JsonApiSerializer.JsonApi;
+using JsonApiSerializer.JsonApi.WellKnown;
+using JsonApiSerializer.JsonConverters;
+using JsonApiSerializer.SerializationState;
+using JsonApiSerializer.Util;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text.RegularExpressions;
+
+namespace JsonApiSerializer.JsonConverters
+{
+    using ContractResolvers.Attributes;
+
+    /// <summary>
+    /// Provides functionality to convert a JsonApi resource object into a .NET object
+    /// </summary>
+    /// <seealso cref="Newtonsoft.Json.JsonConverter" />
+    public class ResourceObjectConverter : JsonConverter
```

Comment:
Git is showing the entire file is different, perhaps different line endings?

Can you fix it up so only the differences are shown, its impossible to see what has changed

Reply:
Interesting, I can try to re-apply my changes, seems some file had LF whereas this one was CRLF